pkg/pod/entrypoint_lookup_impl.go

```diff
@@ -67,7 +67,13 @@ func (e *entrypointCache) Get(ref name.Reference, namespace, serviceAccountName
 		return nil, fmt.Errorf("error creating k8schain: %v", err)
 	}
 	mkc := authn.NewMultiKeychain(kc)
-	img, err := remote.Image(ref, remote.WithAuthFromKeychain(mkc))
+	//By default go-containerregistry pulls amd64 images.
+	//Setting correct image pull architecture based on the underlying platform
+	var pf = v1.Platform {
+        		Architecture: runtime.GOARCH ,
+	        	OS :          "linux",
+        	}
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(mkc), remote.WithPlatform(pf))
```

Comment:
This is probably the right band-aid for supporting multi-arch on homogeneous clusters short-term, but we should probably open an issue to further discuss heterogeneous clusters (where the controller may be running on a different platform than the pod gets scheduled on).

One problem with this in general is that different architectures could (technically) have different entrypoints, so to properly support multi-arch on heterogeneous clusters, I think we will need to pass through all of the arch-specific configs and have the endpoint select the appropriate one based on where it schedules.

Reply:
To give an example of a subtle, but significant variation in config, when playing with multi-arch `kaniko` I rebuilt with Bazel and ended up with `/kaniko/executor_{arch}` as the entrypoint in the respective architecture images.  So while my early intuition was that these should largely be the same, I don't think we can assume that.

Again: this probably shouldn't hold up this tactical change for homogeneous clusters, but is something that should be considered for heterogeneous multi-arch support.

Reply:
@mattmoor right, but we *do* fetch the image here to get the entrypoint from the config. So if the entrypoint is correctly set in the image to `/kaniko/executor_{arch}`, it shouldn't be a problem, should it ? :thinking:

Reply:
> it shouldn't be a problem, should it ? 

Only if the arch where the pod will be scheduled matches the arch of the node where this code is running to look up the entrypoint. If you have a controller on amd64 and schedule a task on arm, you could run into issues (assuming I understand correctly what this code is doing).

Reply:
> If you have a controller on amd64 and schedule a task on arm, you could run into issues (assuming I understand correctly what this code is doing).
Ah, indeed, I didn't thought of that one :sweat:

Reply:
I think it's worth calling out this behavior in a comment in the code. Something like:

```
// By default go-containerregistry pulls amd64 images.
// Setting correct image pull architecture based on the underlying platform
// _of the node that Tekton's controller is running on_. If the cluster
// is comprised of nodes of heterogeneous architectures, this might cause issues.
```

Reply:
Make a suggested edit with this above, since I was adding a nit for spacing anyhow.